### ISSUE TITLE
Padding on ImageButtons works wrong on Android

### ISSUE THREAD

**Issue by matzeteo** [dotnet/maui#13101]
### Description

If you add padding to an ImageButton on Android, not only a padding will be set, but also a margin with the size of the padding. The button will therefore become smaller due to the padding. To get the button back to the right size, you then need to set a margin with a negative size.

### Steps to Reproduce

1. Create a new Maui App
2. Create a Grid
3. Add a ImageButton to the Grid
4. Set a Padding on the ImageButton
5. Add a Button to the Grid to see the Size difference
6. It should be now like:
```xaml
 <Grid ColumnDefinitions="*,100,100,*" RowDefinitions="*,100,*">
        <ImageButton Grid.Column="1" Grid.Row="1" Padding="10" BackgroundColor="Aqua" Source="dotnet_bot.png"/>
        <Button Grid.Column="2" Grid.Row="1" Text="testButton" Padding="10" BackgroundColor="Blue" WidthRequest="100" HeightRequest="100"/>
    </Grid>
```
### Link to public reproduction project repository

https://github.com/matzeteo/TestApps.git

### Version with bug

8.0 (current)

### Affected platforms

Android

### Affected platform versions

Android 10.0 and up

### Did you find any workaround?

Set a margin with the negative value of the padding. But that doesn't work for every Layout structure.

**Comment by AswinPG** [comment]
https://github.com/dotnet/maui/issues/7927

**Comment by XamlTest** [comment]
Verified this on Visual Studio Enterprise 17.7.0 Preview 2.0. Repro on Android 13.0-API33 with below Project:
[TestApp.zip](https://github.com/dotnet/maui/files/11797197/TestApp.zip)

![image](https://github.com/dotnet/maui/assets/97647558/e298f88c-7303-411d-9faf-bea2d7f647fa)

**Comment by jsuarezruiz** [comment]
This issue should have been fixed by https://github.com/dotnet/maui/pull/14905
<img width="456" alt="Captura de pantalla 2023-06-20 a las 10 15 14" src="https://github.com/dotnet/maui/assets/6755973/be2fb81b-846c-4bef-a1fc-7336deb4ed9d">

**Comment by samtun** [comment]
We found a workaround for our use case by using Aspect="Center" for the ImageButton which scales correctly for only a simple image inside an ImageButton. Not sure if this is a fix for any use case but I guess it's worth trying.

**Comment by awasilik** [comment]
I can still observe the issue with newest version of Maui (8.0.10)

**Comment by gautambjain** [comment]
Wow @samtun - Aspect="Center" works. This is a good workaround. Thanks for sharing.

@jsuarezruiz This issue still exists in Maui 8.0.10.

**Comment by FrancescoCarraro** [comment]
I'm trying to use it with Maui 8 but Padding works only on Windows, not on Android

**Comment by MPoels** [comment]
It's still not working on Maui 8.0.21 either.  (Aspect="Center" is not working as a workaround)
If I modify it at runtime within the debugger at the emulator it works.

Hopefully this is fixed soon, a very essential one to get your imagebutton look well. Probably fixable by making a customhandler but it should work out of the box.

**Comment by BlueRaja** [comment]
Aspect="Center" does not workaround the bug.  Rather, _any_ change to the button at runtime will cause the XAML hot-reload to refresh the button, which then causes it to render correctly (!?!).  However, after restarting the app, it's broken again.

This bug was supposed to be fixed in https://github.com/dotnet/maui/pull/14905 but it seems to have broken again. The PR https://github.com/dotnet/maui/pull/22298 supposedly fixes it again.  This latter PR is tagged as both "fixed in 8.0.60" and "fixed in 9.0.0".  The bug still exists in 8.0.80, so clearly it was not fixed in 8.0.60.  Hopefully it's _actually_ fixed in 9.0.0.........

**Comment by xibeat** [comment]
Very sad to see this state of basic controls not working properly in Maui. We are creating an app since last September and failed to ship an MVP because of cosmetic issues like this. Maui is a huge let down at this point. We will wait for .NET 9 at best and then we will never return to Maui. Thanks for this pathetic experience team :-)

**Comment by evgenyvalavin** [comment]
+1

**Comment by evgenyvalavin** [comment]
Still not fixed in .NET 9

**Comment by BlueRaja** [comment]
Yep, confirmed that padding is still broken in the [same way](https://github.com/BlueRaja) in .Net 9.

CC @mattleibow since you worked on the PR that was supposed to fix this 😕

**Comment by BlueRaja** [comment]
Just to update, the workaround posted above doesn't work on .Net 9.  Or rather, it works about 25% of the time, seemingly at random.  Changing the wait time does not change this.

**Comment by sheiksyedm** [comment]
The PR linked with this issue has been merged into the main branch via PR https://github.com/dotnet/maui/pull/28560. Therefore, this issue can now be closed.
